src/control/jobs/XournalScheduler.cpp

```diff
@@ -12,51 +12,60 @@ XournalScheduler::XournalScheduler()
 XournalScheduler::~XournalScheduler() = default;
 
 void XournalScheduler::removeSidebar(SidebarPreviewBaseEntry* preview) {
-    // Dont await for tasks, as it completly blocks whole program.
-    // Not sure why.
-    removeSource(preview, JOB_TYPE_PREVIEW, JOB_PRIORITY_HIGH, false);
+    removeSource(preview, JOB_TYPE_PREVIEW, JOB_PRIORITY_HIGH, true);
 }
 
 void XournalScheduler::removePage(XojPageView* view) { removeSource(view, JOB_TYPE_RENDER, JOB_PRIORITY_URGENT); }
 
 void XournalScheduler::removeAllJobs() {
-    g_mutex_lock(&this->jobQueueMutex);
+    std::lock_guard<std::mutex> lock(this->jobQueueMutex);
```

Comment:
**nitpick:** Very good change :+1: But it's not required in cpp17 to write the template parameter, also use `{}` to initialize objects.
```suggestion
    std::unique_lock guard{this->jobQueueMutex};
```

Reply:
Nice catch!